### PR TITLE
Image stream

### PR DESCRIPTION
**Pull request by navidsh**
**What this PR does / why we need it?**:
- Supports referencing an Image Stream Tag in CR
- Watches Image Stream changes and reconciles applications using the image stream
- Upgraded `operator-sdk` to version `0.15.2`

**Does this PR introduce a user-facing change?**

- [x] User guide
- [x] `CHANGELOG.md`

### CODE REVIEW

**Comment by navidsh** [comment]
Fixes #115

**Comment by navidsh** [comment]
I will address the review comments in a separate PR to avoid merge conflict with another ongoing PR.

File: doc/user-guide.md [hunk 1]
```diff
@@ -121,6 +123,28 @@ oc get appsodyapplication my-appsody-app
...
+spec:
+  stack: java-microprofile
+  applicationImage: my-namespace/my-image-stream:1.0
+```
+
+The above example will look up the `1.0` tag from the `my-image-stream` image stream in `my-namespace` project and populate the CR's `.status.imageReference` field with the exact referenced image similar to the following: `image-registry.openshift-image-registry.svc:5000/my-namespace/my-image-stream@sha256:8a829d579b114a9115c0a7172d089413c5d5dd6120665406aae0600f338654d8`. The operator keeps watching the specified image stream and will deploy new images as new ones are available for the specified tag.
+
+To reference an image stream, the `applicationImage` should follow `<project name>/<image stream name>[:<tag>]` format. If `<project name>` or `<tag>` is not specified, the operator default the values to CR's namespace and `latest`, respectively. For example `applicationImage: my-image-stream` would be the same as `my-namespace/my-image-stream:latest`.
+
+The Operator would try to match an image stream name first with `<project name>/<image stream name>[:<tag>]` format and fallback into registry lookup if it is not able to find any image stream that matches the value.
```
Comment (chirp1) [inline doc/user-guide.md]: Change :The Operator would try to match an image stream name first with `<project name>/<image stream name>[:<tag>]` format and fallback into registry lookup if it is not able to find any image stream that matches the value. " to "The Operator **tries** to match an image stream name first with **the** `<project name>/<image stream name>[:<tag>]` format and **falls back to the** registry lookup if it is not able to find any image stream that matches the value. "

File: doc/user-guide.md [hunk 2]
```diff
@@ -121,6 +123,28 @@ oc get appsodyapplication my-appsody-app
...
+metadata:
+  name: my-appsody-app
+spec:
+  stack: java-microprofile
+  applicationImage: my-namespace/my-image-stream:1.0
+```
+
+The above example will look up the `1.0` tag from the `my-image-stream` image stream in `my-namespace` project and populate the CR's `.status.imageReference` field with the exact referenced image similar to the following: `image-registry.openshift-image-registry.svc:5000/my-namespace/my-image-stream@sha256:8a829d579b114a9115c0a7172d089413c5d5dd6120665406aae0600f338654d8`. The operator keeps watching the specified image stream and will deploy new images as new ones are available for the specified tag.
+
+To reference an image stream, the `applicationImage` should follow `<project name>/<image stream name>[:<tag>]` format. If `<project name>` or `<tag>` is not specified, the operator default the values to CR's namespace and `latest`, respectively. For example `applicationImage: my-image-stream` would be the same as `my-namespace/my-image-stream:latest`.
```
Comment (chirp1) [inline doc/user-guide.md]: Change "the `applicationImage` should follow `<project name>/<image stream name>[:<tag>]` format." to "the `applicationImage` should follow `<project name>/<image stream name>[:<tag>]` format." to "the `applicationImage` **parameter must**  follow **the** `<project name>/<image stream name>[:<tag>]` format.".
Change "If `<project name>` or `<tag>` is not specified, the operator default the values to CR's namespace and `latest`, respectively." to "If the `<project name>` or `<tag>` variable is not specified, the operator **defaults** the values to **the namespace of the CR and the value of `latest`**, respectively. " (If "variables" isn't the correct noun, replace it with the correct noun.)

File: doc/user-guide.md [hunk 3]
```diff
@@ -44,14 +44,14 @@ spec:
...
-Each `AppsodyApplication` CR must specify `applicationImage` parameter. Specifying other parameters is optional.
+Each `AppsodyApplication` CR must at least specify the `applicationImage` parameter. Specifying other parameters is optional.
 
 | Parameter | Description |
 |---|---|
 | `stack` | The name of the Appsody Application Stack that produced this application image. |
 | `version` | The current version of the application. Label `app.kubernetes.io/version` will be added to all resources when the version is defined. |
 | `serviceAccountName` | The name of the OpenShift service account to be used during deployment. |
-| `applicationImage` | The absolute name of the image to be deployed, containing the registry and the tag. |
+| `applicationImage` | The Docker image name to be deployed. On OpenShift, it can also be set to `<project name>/<image stream name>[:<tag>]` to reference an image from an image stream. `<project name>` and `tag` defaults to CR's namespace and `latest` if not defined, respectively. |
```
Comment (chirp1) [inline doc/user-guide.md]: Change "`<project name>` and `tag` defaults to CR's namespace and `latest` if not defined, respectively." to "The `<project name>` and `tag` variables default to the namespace of the CR and the value of `latest` respectively, if they are not defined.".

**Pull request merged**